ssl/ssl_lib.c

```diff
@@ -5913,7 +5920,10 @@ const EVP_MD *ssl_evp_md_fetch(OPENSSL_CTX *libctx,
 #endif
 
     /* Otherwise we do an explicit fetch */
-    return EVP_MD_fetch(libctx, OBJ_nid2sn(nid), properties);
+    ERR_set_mark();
+    md = EVP_MD_fetch(libctx, OBJ_nid2sn(nid), properties);
+    ERR_pop_to_mark();
+    return md;
```

Comment:
If `md` is NULL at this point, is that an error or not?  If it is, `ERR_set_mark()` / `ERR_pop_to_mark()` is wrong

Reply:
No. It's not an error. Sometimes we attempt to get an MD that isn't available everywhere (e.g. GOST mds). We don't know if it's there until we try and get hold of it. If this function returns NULL we just ignore the digest and move on to the next one.

Reply:
Okie, fair enough